src/main/java/com/naijagov/naijagov/controller/NaijagovController.java

```diff
@@ -21,89 +21,115 @@
 import org.springframework.web.servlet.ModelAndView;
 
 @Controller
-@SessionAttributes({"form"})
-public class NaijagovController{
-
-    @GetMapping("/wards")
-    public ModelAndView wards(@PageableDefault(size=10, sort="fullCode", 
-                                direction=Sort.Direction.ASC) Pageable pageable, 
-                                @ModelAttribute Form form, ModelAndView mv, 
-                                @RequestParam("sortBy") Optional<String> sortBy, 
-                                @RequestParam("sortDirection") Optional<Sort.Direction> sortDirection){
-        Pageable sortedPage = pageable;
-        //initial page load, no params
-        if(form.getWardsPage()==null){
-            sortedPage=pageable;
-        }
-        //when header sort is clicked
-        else if(sortBy.isPresent() &&sortDirection.isPresent()) 
-            sortedPage = PageRequest.of(form.getWardsPage().getNumber(), 
-                                      form.getWardsPage().getSize(),
-                                      sortDirection.get(), sortBy.get());
-        //when navigation link is clicked
-        else if(form.getWardsPage()!=null) 
-            sortedPage = PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), 
-                                        form.getWardsPage().getSort());
-        form.setWardsPage(wardService.listAllWards(sortedPage));
-
-        mv.setViewName("wards");
-        return mv;
+@SessionAttributes({ "form" })
+public class NaijagovController {
+
+  private PollingUnitService pollingUnitService;
+  private WardService wardService;
+
+  @GetMapping("/wards")
+  public ModelAndView wards(
+      @PageableDefault(size = 10, sort = "fullCode", direction = Sort.Direction.ASC) Pageable pageable,
+      @ModelAttribute Form form, ModelAndView mv, @RequestParam("sortBy") Optional<String> sortBy,
+      @RequestParam("sortDirection") Optional<Sort.Direction> sortDirection) {
+    Pageable sortedPage = pageable;
+    // initial page load, no params
+    if (form.getWardsPage() == null) {
+      sortedPage = pageable;
     }
+    // when header sort is clicked
+    else if (sortBy.isPresent() && sortDirection.isPresent())

```

Comment:
I recommend always making your blocks with curly braces , it makes your code a lot easier to read

Reply:
Fixed in commit [bc618b2](a1e795154d7813c8448dfd349695fe3b0ddcf088)